cairo/ethereum/cancun/trie.cairo

```diff
@@ -133,6 +166,91 @@ struct Subnodes {
     value: SubnodesStruct*,
 }
 
+func Subnodes__eq__(left: Subnodes, right: Subnodes) -> bool {
+    let branch0_eq = Extended__eq__(left.value.branch_0, right.value.branch_0);
```

Comment:
do you want to reuse the good lookin' logic you used below of subnodes?

- an inner function,
- an index to access branch_i